python3Packages.coordinates: init at 0.4.0

mildlyincompetent opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

Packaging coordinates, a python module.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

mildlyincompetent committed:
Add mildlyincompetent to the maintainers list

mildlyincompetent committed:
python3Packages.coordinates: init at 0.4.0

mildlyincompetent commented:
@GrahamcOfBorg build python36Packages.coordinates python37Packages.coordinates python38Packages.coordinates

bcdarwin commented on pkgs/development/python-modules/coordinates/default.nix:
```diff
@@ -0,0 +1,21 @@
+buildPythonPackage rec {
+  pname = "coordinates";
+  version = "0.4.0";
+
```
Please set `disabled = !isPy3k;`.

bcdarwin replied:
Sorry, also needs to be disabled for Python 3.5.

jonringer replied:
`disabled = pythonOlder "3.5";`

bcdarwin commented on pkgs/development/python-modules/coordinates/default.nix:
```diff
@@ -0,0 +1,21 @@
+  };
+
+  # No tests in PyPi source distribution.
+  doCheck = false;
```
I see you've requested these be bundled with the PyPi dist, but possibly consider switching to the GitHub release for now?

bcdarwin reviewed (changes requested):
Thanks!  See a couple comments, mainly disabling for unsupported Pythons.  Python 3 version build on my NixOS machine.

bcdarwin commented:
Also, please update the PR/commit to a buildable package name, here `python3Packages.coordinates: ...`.  Thanks!

jonringer reviewed (commented):
Also, the commit message should be:
```
pythonPackages.coordinates: init at 0.4.0
```

mildlyincompetent commented:
@GrahamcOfBorg build python3Packages.coordinates python36Packages.coordinates python37Packages.coordinates python38Packages.coordinates

mildlyincompetent commented:
Thanks for the advice everyone, I've
- Switched to the github release, as that includes test
- Disabled for python 3.5 and below
- Fixed the commit message

Let me know if there's anything further.

bcdarwin commented on pkgs/development/python-modules/coordinates/default.nix:
```diff
@@ -0,0 +1,30 @@
+  checkInputs = [ pytest ];
+
+  checkPhase = ''
+    pytest tests/
```
should be (not ideal admittedly)

checkPhase = ''
  runHook preCheck
  pytest tests/
  runHook postCheck
'';

jonringer replied:
in most cases, the explicit hooks aren't super necessary

jonringer reviewed (approved):
diff LGTM
commits LGTM
has tests :+1: 

```
[4 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/77335
2 package were built:
python37Packages.coordinates python38Packages.coordinates
```

jonringer commented:
@GrahamcOfBorg build python37Packages.coordinates python38Packages.coordinates
